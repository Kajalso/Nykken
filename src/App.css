.App {
  padding: 0 1em;
}

.sensor-grid {
  display: grid;
  grid-template-columns: auto auto;
  justify-content: start;
  column-gap: 5em;
  row-gap: 2em;
}

h2 {
  font-weight: 400;
}
