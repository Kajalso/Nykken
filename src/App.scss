$primaryText: #3b4541;

.App {
  padding: 0;
}

/* Font styles */
h1,
h2,
h3,
h4,
h5,
h6 {
  font-weight: 400;
  color: $primaryText;
}

h1 {
  font-size: 20pt;
}

h2 {
  font-size: 16pt;
}

h3 {
  font-size: 16pt;
}

h4 {
  font-size: 18pt;
}

h5 {
  font-size: 15pt;
}

h6 {
  font-size: 13pt;
}

p {
  font-size: 12px;
}

p.small {
  font-size: 10px;
}

.button {
  font-size: 12px;
  font-weight: 400;
}

.button.small {
  font-size: 10px;
}
