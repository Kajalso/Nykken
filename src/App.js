import "./App.scss";
import { Header } from "./components/Header/Header";
import { Dashboard } from "./components/Dashboard/Dashboard";
import GroupsContextProvider from "./context/GroupsContext";

function App() {

  return (
    <div className="App">
      <GroupsContextProvider>
      <Header />
      <Dashboard />
      </GroupsContextProvider>
    </div>
  );
}

export default App;
