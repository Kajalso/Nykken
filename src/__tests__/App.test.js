import React from 'react';
import { render } from '@testing-library/react'

import App from '../App';


//Suppressing warning as suggested by:
const originalError = global.console.error;
beforeAll(() => {
	global.console.error = jest.fn((...args) => {
		if (typeof args[0] === 'string' && args[0].includes('Please upgrade to at least react-dom@16.9.0')) {
	      return
	    }
	    return originalError.call(console, args)
	});
});

afterAll(() => {
  	global.console.error.mockRestore();
});

describe('renders without crashing and takes snapshot', () => {
    test('renders correctly', () => {
        render(< App />);
    });
    test('takes snapshot', () => {
        const component = render(< App />);
        expect(component).toMatchSnapshot();
    })
});