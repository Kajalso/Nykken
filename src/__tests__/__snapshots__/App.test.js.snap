// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and takes snapshot takes snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="header"
        >
          <h2>
            Risvollan measurement station
          </h2>
          <p
            class="small"
          >
            Last data update: 05.03.21 06:02
          </p>
        </div>
        <div
          class="dashboard"
        >
          <div
            class="sensor-measurements"
          >
            <h2
              class="title"
            >
              Sensor measurements
            </h2>
            <button
              class="button small"
            >
              <img
                alt="Button icon"
                src="edit.svg"
              />
              Add/remove sensors
            </button>
            <div
              class="sensor-grid"
            >
              <p>
                No sensors chosen
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ReactModalPortal"
    />
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="header"
      >
        <h2>
          Risvollan measurement station
        </h2>
        <p
          class="small"
        >
          Last data update: 05.03.21 06:02
        </p>
      </div>
      <div
        class="dashboard"
      >
        <div
          class="sensor-measurements"
        >
          <h2
            class="title"
          >
            Sensor measurements
          </h2>
          <button
            class="button small"
          >
            <img
              alt="Button icon"
              src="edit.svg"
            />
            Add/remove sensors
          </button>
          <div
            class="sensor-grid"
          >
            <p>
              No sensors chosen
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
